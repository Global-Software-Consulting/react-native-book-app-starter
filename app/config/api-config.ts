/* App config for apis
 */
const ApiConfig = {
  BASE_URL: 'http://wolverine-dev.com/',
  LOGIN: 'api/login',
};

export default ApiConfig;
