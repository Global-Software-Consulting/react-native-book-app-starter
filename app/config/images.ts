/* App config for images
 */
const images = {
  icons: {
    //logo: require('../assets/images/icons/logo.png'),
  },
};

export default images;
