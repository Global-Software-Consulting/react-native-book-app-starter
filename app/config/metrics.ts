/*
 * platform/application wide metrics for proper styling
 */
import { Dimensions, Platform } from 'react-native';
const { width, height } = Dimensions.get('window');

const metrics = {
  screenWidth: width < height ? width : height,
  screenHeight: width < height ? height : width,
  navBarHeight: Platform.OS === 'ios' ? 54 : 66,
};

export default metrics;
