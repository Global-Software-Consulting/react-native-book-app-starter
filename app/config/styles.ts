/*
 * Provides universal color configs used in the app.
 * Provides universal fonts used in the app.
 */
const AppStyles = {
    colors: {
        COLOR_PRIMARY: '#2ec7ab',
        COLOR_SECONDARY: '#111',
        COLOR_WHITE: '#FFFFFF',
        COLOR_BLACK: '#000000',
        COLOR_GREY: 'grey',
        COLOR_GREEN: 'green',
        COLOR_PLACEHOLDER: '#111111',
        COLOR_GREY_WHITE: '#fafafa',
        COLOR_DARK_SEPERATOR: '#d4d4d4',
        COLOR_BLACK_TRANSP: 'rgba(0, 0, 0, 0.7)',
        COLOR_GREY_TRANSP: 'rgba(67, 85, 85, 0.7)',
    },
    fonts: {
        // FONT_REGULAR: 'Roboto-Regular',
        // FONT_MEDIUM: 'Roboto-Medium'
    },
};
export default AppStyles;
