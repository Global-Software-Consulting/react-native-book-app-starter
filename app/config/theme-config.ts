import { DarkTheme as PaperDarkTheme, DefaultTheme as PaperDefaultTheme } from 'react-native-paper';

export const PaperThemeDefault = {
    ...PaperDefaultTheme,
    colors: {
        ...PaperDefaultTheme.colors,
        primary: '#3498db',
        iconColor: 'blue',
    },
};

export const PaperThemeDark = {
    ...PaperDarkTheme,
    colors: {
        ...PaperDarkTheme.colors,
        primary: '#3498db',
        iconColor: 'white',
    },
};
