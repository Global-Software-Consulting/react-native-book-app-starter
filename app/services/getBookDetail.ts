import ApiConfig from '../config/api-config';
import Api from './client';

export default function getBookDetail(id: number) {
    return Api(ApiConfig.BASE_URL + ApiConfig.BOOKDETAIL + id, null, 'GET');
}
