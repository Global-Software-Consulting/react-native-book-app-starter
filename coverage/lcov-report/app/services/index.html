
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for app/services</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> app/services</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">3.12% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>1/32</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">0% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/13</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">0% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>0/17</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">3.12% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>1/32</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="addBookToFavoite.ts"><a href="addBookToFavoite.ts.html">addBookToFavoite.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	</tr>

<tr>
	<td class="file low" data-value="client.ts"><a href="client.ts.html">client.ts</a></td>
	<td data-value="4.54" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 4%"></div><div class="cover-empty" style="width: 96%"></div></div>
	</td>
	<td data-value="4.54" class="pct low">4.54%</td>
	<td data-value="22" class="abs low">1/22</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="13" class="abs low">0/13</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="4.54" class="pct low">4.54%</td>
	<td data-value="22" class="abs low">1/22</td>
	</tr>

<tr>
	<td class="file low" data-value="forgotPassword.ts"><a href="forgotPassword.ts.html">forgotPassword.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	</tr>

<tr>
	<td class="file low" data-value="getBookDetail.ts"><a href="getBookDetail.ts.html">getBookDetail.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	</tr>

<tr>
	<td class="file low" data-value="getBooks.ts"><a href="getBooks.ts.html">getBooks.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	</tr>

<tr>
	<td class="file low" data-value="getFavoriteBooks.ts"><a href="getFavoriteBooks.ts.html">getFavoriteBooks.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	</tr>

<tr>
	<td class="file low" data-value="getUserDetail.ts"><a href="getUserDetail.ts.html">getUserDetail.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	</tr>

<tr>
	<td class="file low" data-value="loginUser.ts"><a href="loginUser.ts.html">loginUser.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	</tr>

<tr>
	<td class="file low" data-value="removeBookFromFavoite.ts"><a href="removeBookFromFavoite.ts.html">removeBookFromFavoite.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	</tr>

<tr>
	<td class="file low" data-value="signupUser.ts"><a href="signupUser.ts.html">signupUser.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	</tr>

<tr>
	<td class="file low" data-value="updateProfile.ts"><a href="updateProfile.ts.html">updateProfile.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Wed Nov 10 2021 14:36:08 GMT+0500 (Pakistan Standard Time)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    