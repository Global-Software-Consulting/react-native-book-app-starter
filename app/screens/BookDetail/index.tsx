import { useIsFocused } from '@react-navigation/core';
import React, { useEffect, useState } from 'react';
import { View } from 'react-native';
import Toast from 'react-native-simple-toast';
import { useDispatch } from 'react-redux';
import getBookDetail from 'services/getBookDetail';
//importing components
import Container from './Container';
import Shimmer from './Shimmer';
import { Books, RouteState } from './types';
const base_url: string = 'https://ebook-application.herokuapp.com/v1/';

const BookDetail: React.FC<RouteState> = (props) => {
    const dispatch = useDispatch();
    const bookId = props.route.params; //getting routed params
    const [bookData, setBookData] = useState<Books>();
    const [isLoading, setIsLoading] = useState(true);
    const isFocussed = useIsFocused();

    //handling back hardware button
    useEffect(() => {
        setIsLoading(true);
        try {
            getBookDetail(parseInt(bookId, 10)).then((response) => {
                if (response.status !== 'success' || response.result !== null) {
                    setBookData(response.result);
                    setIsLoading(false);
                } else {
                    Toast.show('Error loading data', Toast.SHORT);
                }
            });
        } catch {
            Toast.show('Error occured', Toast.SHORT);
        }
    }, [bookId, dispatch, isFocussed]);

    return (
        <View>
            {isLoading || bookData === undefined || bookData === null ? (
                <Shimmer />
            ) : (
                <Container base_url={base_url} books={bookData} />
            )}
        </View>
    );
};

export default BookDetail;
