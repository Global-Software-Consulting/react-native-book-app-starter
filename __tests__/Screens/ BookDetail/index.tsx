import { useIsFocused } from '@react-navigation/core'
import { render } from '@testing-library/react-native'
import React from 'react'
import * as getDetail from 'services/getBookDetail'

import * as redux from 'react-redux'
import { act } from 'react-test-renderer'
import BookDetail from 'screens/BookDetail/index'
import {} from './../../../app/config/images'
import { useDispatch } from 'react-redux'
jest.useFakeTimers
jest.mock('./../../../app/services/getBookDetail', () => {
    return {
        getBookDetail: jest.fn({ status: 'notSucess', result: 'ds' }),
    }
})

jest.mock('./../../../app/screens/BookDetail/types', () => {
    return {
        props: { route: { params: jest.fn() } },
    }
})

jest.mock('@react-navigation/core', () => {
    return {
        useIsFocused: jest.fn(),
    }
})

jest.mock('react-native-image-crop-picker', () => {
    return {
        __esModule: true,
        A: true,
        useDispatch: jest.fn(),
        default: 'mockedDefaultExport',
    }
})

jest.mock('react-native-keyboard-aware-scroll-view', () => {
    return {
        __esModule: true,
        A: true,
        KeyboardAwareScrollView: jest.fn(),
        default: 'mockedDefaultExport',
    }
})
jest.mock('react-native-linear-gradient', () => {
    return {
        __esModule: true,
        A: true,
        LinearGradient: jest.fn(),
        default: 'mockedDefaultExport',
    }
})
jest.mock('./../../../app/config/images', () => {
    return {
        __esModule: true,
        A: true,
        app: { logo: jest.fn() },
        default: 'mockedDefaultExport',
    }
})

jest.mock('react-native-tts', () => {
    return {
        __esModule: true,
        A: true,
        Tts: jest.fn(),
        default: 'mockedDefaultExport',
    }
})

describe('Component testing', () => {
    beforeAll(() => {
        const dispatch = jest.fn()
        useDispatch.mockReturnValue(jest.fn())
    })
    test('Snapshot ', () => {
        let state = { params: 'abc' }
        const tree = render(<BookDetail route={state} />)
        state = { params: 'cde' }
        expect(tree).toMatchSnapshot()
    })

    test('Service test with success response', () => {
        let response = { status: 'success' }
        const mockSendOtpCodeByUserCode = jest
            .spyOn(getDetail, 'getBookDetail')
            .mockImplementation(() => Promise.resolve(responseMock))
        const tree = render(<BookDetail />)
    })
})
