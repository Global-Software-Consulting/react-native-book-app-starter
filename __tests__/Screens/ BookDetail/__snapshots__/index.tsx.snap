// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component testing Snapshot  1`] = `
<View>
  <View>
    <RCTScrollView
      contentContainerStyle={
        Object {
          "alignItems": "center",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#E7E5E7",
              "borderRadius": 20,
              "elevation": 5,
              "height": 400,
              "margin": 10,
              "width": 375,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#E7E5E7",
                "height": 53.5,
                "marginRight": 15,
                "width": 300,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#E7E5E7",
                "height": 53.5,
                "marginRight": 15,
                "width": 300,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#E7E5E7",
              "height": 133.5,
              "margin": 10,
              "width": 600,
            }
          }
        />
        <View
          style={
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#E7E5E7",
              "height": 53.5,
              "marginLeft": 67.5,
              "marginRight": 15,
              "width": 300,
            }
          }
        />
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#E7E5E7",
              "height": 267,
              "margin": 5,
              "width": 600,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`Component testing Snapshot  2`] = `
<View>
  <View>
    <RCTScrollView
      contentContainerStyle={
        Object {
          "alignItems": "center",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#E7E5E7",
              "borderRadius": 20,
              "elevation": 5,
              "height": 400,
              "margin": 10,
              "width": 375,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#E7E5E7",
                "height": 53.5,
                "marginRight": 15,
                "width": 300,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#E7E5E7",
                "height": 53.5,
                "marginRight": 15,
                "width": 300,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#E7E5E7",
              "height": 133.5,
              "margin": 10,
              "width": 600,
            }
          }
        />
        <View
          style={
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#E7E5E7",
              "height": 53.5,
              "marginLeft": 67.5,
              "marginRight": 15,
              "width": 300,
            }
          }
        />
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#E7E5E7",
              "height": 267,
              "margin": 5,
              "width": 600,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
</View>
`;
